spec/rubocop/cop/rspec/capybara/specific_matcher_spec.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::Capybara::SpecificMatcher do
+  it 'does not register an offense for abstract matcher when elements with' \
+    'no matcher specified in detail' do
+    expect_no_offenses(<<-RUBY)
+      expect(page).to have_selector('article')
+      expect(page).to have_none_of_selectors('body')
+      expect(page).to have_css('tbody')
+    RUBY
+  end
+
+  it 'does not register an offense for abstract matcher when elements with' \
+    'not element' do
+    expect_no_offenses(<<-RUBY)
+      expect(page).to have_no_css('.a')
+      expect(page).to has_css?('#button')
+      expect(page).to has_no_css?('[table]')
+    RUBY
+  end
+
+  it 'registers an offense when using `#have_selector`' do
+    expect_offense(<<-RUBY)
+    expect(page).to have_selector('button')
+                    ^^^^^^^^^^^^^^^^^^^^^^^ Use `#have_button` instead of `#have_selector`.
+    expect(page).to have_selector('a')
+                    ^^^^^^^^^^^^^^^^^^ Use `#have_link` instead of `#have_selector`.
+    expect(page).to have_selector('table')
+                    ^^^^^^^^^^^^^^^^^^^^^^ Use `#have_table` instead of `#have_selector`.
+    expect(page).to have_selector('select')
+                    ^^^^^^^^^^^^^^^^^^^^^^^ Use `#have_select` instead of `#have_selector`.
+    RUBY
+  end
+
+  it 'registers an offense when using `#have_none_of_selectors`' do
+    expect_offense(<<-RUBY)
+    expect(page).to have_none_of_selectors('button')
+                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `#have_no_button` instead of `#have_none_of_selectors`.
```

Comment:
Is there a `have_no_button`?

Reply:
I intended what is defined here `have_no_button`.
https://github.com/teamcapybara/capybara/blob/51bca8300136ba2dcf45e92883a07845105f2b3a/lib/capybara/rspec/matchers.rb#L162-L169